access.lua

```diff
@@ -55,30 +82,34 @@ end
 -- ###########################################################################
 --     1. AUTHENTICATION
 --  Check wether or not this is a logged-in user
+--  This is not run immediately but only if:
+--  - the app is not public
+--  - and/or auth_headers is enabled for this app
 -- ###########################################################################
 
 function check_authentication()
 
     -- cf. src/authenticators/ldap_ynhuser.py in YunoHost to see how the cookie is actually created
 
     local cookie = ngx.var["cookie_" .. conf["cookie_name"]]
+    if cookie == nil then
+        return false, nil, nil
+    end
 
-    decoded, err = jwt.verify(cookie, "HS256", cookie_secret)
+    user, pwd, err = cached_jwt_verify(cookie, "H256", cookie_secret)
```

Comment:
```suggestion
    user, pwd, err = cached_jwt_verify(cookie, cookie_secret)
```

(cf previous comment about changing the signature of `cached_jwt_verify`)